db/version_set.cc

```diff
@@ -2833,6 +2833,8 @@ Status Version::MultiGetAsync(
         std::vector<Status> statuses = folly::coro::blockingWait(
             folly::coro::collectAllRange(std::move(mget_tasks))
                 .scheduleOn(&range->context()->executor()));
+        assert(mget_tasks.size() == 0);
+        mget_tasks.clear();
```

Comment:
Is this fix to clear the `mget_tasks`. Won't it give an assert issue if it contains the data before clearing?

Reply:
The clear is just in case, in a non-debug build. Practically speaking, I expect the move to always clear the vector. This was flagged as a potential bug by Phabricator based on static analysis.

Reply:
Nvm. This didn't silence the static analyzer.

Reply:
According to cppreference, the moved-from vector is not guaranteed to be emptied (though any reasonable implementation *should* empty it), but is guaranteed to be in a valid state. (Therefore, just `clear()` should minimize both false positive and false negative reports of a problem.)